packages/itwin/tree-widget/src/components/trees/repositories-tree/RepositoriesService.ts

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * @internal
+ */
+export interface ITwinRepositories {
+  repositories: ITwinRepository[];
+}
+
+interface Repository {
+  id?: string;
+  displayName?: string;
+  type?: string;
+  name?: string;
+}
+
+interface ITwinRepository {
+  id?: string;
+  class: string;
+  subClass?: string;
+  uri?: string;
+}
```

Comment:
I think we need to be more strict about these types.

For `ITwinRepository`, I think we should make `class` and `uri` required. `id` can probably be removed altogether.

For `Repository` (needs a rename), I think `id` and `displayName` should be required. Not sure what `name` means or how it's different from `displayName` - do we need it?

Reply:
All the repositories have a `displayName` except for the cesium one, which has a name instead, to not add a separate type added both displayName and name.

Reply:
> I think we need to be more strict about these types.
> 
> For `ITwinRepository`, I think we should make `class` and `uri` required. `id` can probably be removed altogether.
> 
> For `Repository` (needs a rename), I think `id` and `displayName` should be required. Not sure what `name` means or how it's different from `displayName` - do we need it?

If we want to, we can be very strict about the types. There are some packages like zod https://zod.dev/,  that help enforce specific rules using regex  (This is out of scope for this PR, but just wanted to mention that).

Reply:
What we store in these objects doesn't necessarily has to map 1:1 to what the service returns. I don't really like the idea of `fetchData` doing some magic like this:

```ts
    const result = await response.json();
    // get the data avoiding links
    delete result._links;
    const values = Object.values(result);
    const data = values[0];

    return data;
```

IMO, it should just return the JSON of `unknown` type. Also, since the request and response formats slightly differ between each service, I'd say we should have a function per service to supply proper headers and parse the response. So for example:

```ts
export async function getIModels(accessToken: string, url: string): Promise<RepositoryData[]> {
  const response = await fetchData({
    url, 
    accessToken,
    extraHeaders: {
      "Accept": "application/vnd.bentley.itwin-platform.v2+json",
      "Prefer": "return=representation",
    },
  });
  if (typeof response !== "object" || !("iModels" in response)) {
    throw new Error("Invalid response");
  }
  const imodels = response["iModels"];
  if (!Array.isArray(imodels)) {
    throw new Error("Invalid response");
  }
  return imodels.map((imodel) => {
    // convert `imodel` to `RepositoryData`.
  });
}
```

I guess @JonasDov suggestion to use a library like `zod` could be helpful here.

Reply:
My idea was to avoid having 8-9 functions that would call for repository data while we are still in the investigation stage.